src/MagentoHackathon/Composer/Magento/Plugin.php

```diff
@@ -79,6 +80,16 @@ class Plugin implements PluginInterface, EventSubscriberInterface
     protected $entryFactory;
 
     /**
+     * @var EventManager
+     */
+    protected $eventManager;
+
+    /**
+     * @var ModuleManager
+     */
+    protected $moduleManager;
```

Comment:
Same here: make it private. But this time make a _protected_ getter, which means you can mock it. So you wouldn't have to use reflection. For the mock to work, you'll have to use the _getter_ throughout this class, instead of accessing the attribute directly. Just like `getEventManager`.